Client/Frontend/Browser/Search/Recent Search/RecentSearchHeaderView.swift

```diff
@@ -0,0 +1,192 @@
+// Copyright 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import UIKit
+import BraveUI
+import Shared
+
+class RecentSearchHeaderView: UICollectionReusableView {
+    private struct DesignUX {
+        static let buttonHeight: CGFloat = 40.0
+        static let paddingX: CGFloat = 15.0
+        static let paddingY: CGFloat = 10.0
+    }
+    
+    private var showRecentSearches = false
+    
+    private let titleLabel = UILabel().then {
+        $0.text = Strings.recentSearchSectionTitle
+        $0.font = .systemFont(ofSize: 17.0, weight: .semibold)
+        $0.textColor = .braveLabel
+    }
+    
+    private let subtitleLabel = UILabel().then {
+        $0.text = Strings.recentSearchSectionDescription
+        $0.font = .systemFont(ofSize: 13.0)
+        $0.textColor = .braveLabel
+        $0.numberOfLines = 0
+        $0.lineBreakMode = .byCharWrapping
+        $0.setContentHuggingPriority(.required, for: .vertical)
+        $0.setContentCompressionResistancePriority(.required, for: .vertical)
+    }
+    
+    let showButton = UIButton()
+    let hideClearButton = UIButton()
+    
+    private let vStackView = UIStackView().then {
+        $0.axis = .vertical
+    }
+    
+    private let hStackView = UIStackView().then {
+        $0.spacing = 9.0
+    }
+    
+    private let hButtonStackView = UIStackView().then {
+        $0.spacing = 9.0
+        $0.setContentHuggingPriority(.required, for: .horizontal)
+    }
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        addSubview(vStackView)
+        vStackView.snp.makeConstraints {
+            $0.leading.equalToSuperview().inset(12.0)
+            $0.trailing.equalToSuperview().inset(12.0)
+            $0.top.bottom.equalToSuperview()
+        }
+        
+        resetLayout(showRecentSearches: showRecentSearches)
+    }
+    
+    @available(*, unavailable)
+    required init(coder: NSCoder) {
+        fatalError()
+    }
+    
+    func resetLayout(showRecentSearches: Bool) {
+        self.showRecentSearches = showRecentSearches
+        themeViews()
+        doLayout()
+    }
+    
+    private func doLayout() {
+        vStackView.arrangedSubviews.forEach({ $0.removeFromSuperview() })
+        hStackView.arrangedSubviews.forEach({ $0.removeFromSuperview() })
+        
+        let spacer = UIView().then {
+            $0.setContentHuggingPriority(.defaultLow, for: .horizontal)
+            $0.setContentCompressionResistancePriority(.required, for: .horizontal)
+        }
+        
+        if showRecentSearches {
+            vStackView.addArrangedSubview(hStackView)
+            
+            [titleLabel, spacer, showButton, hideClearButton].forEach({
+                self.hStackView.addArrangedSubview($0)
```

Comment:
one trick here, you can call it like
```
[titleLabel, spacer, showButton, hideClearButton].forEach(hStackView.addArrangedSubview(_:))
```

it's a bit simpler, can do that in other places too if it looks readable to you